contrib/fparser/fparser_ad.cc

```diff
@@ -111,6 +114,95 @@ Value_t FunctionParserADBase<Value_t>::fp_plog(const Value_t * params)
   return x < a ? fp_log(a) + (x-a)/a - (x-a)*(x-a)/(Value_t(2)*a*a) + (x-a)*(x-a)*(x-a)/(Value_t(3)*a*a*a) : fp_log(x);
 }
 
+template <typename Value_t>
+Value_t FunctionParserADBase<Value_t>::fp_erf(const Value_t * params)
+{
+  const Value_t x = params[0];
+  return std::erf(x);
+}
+
+// Derivation of these complex method can be found here: https://math.stackexchange.com/a/712568/408963
+
+template<>
+std::complex<double>
+FunctionParserADBase<std::complex<double>>::fp_erf(const std::complex<double> * params)
+{
+  const std::complex<double> z = params[0];
+  double x = z.real(), y = z.imag();
+  double fx = std::erf(x) + std::exp(-(x*x)) / (2. * M_PI) * (1. - std::cos(2. * x * y));
+  double fy = std::exp(-(x*x)) / (2. * M_PI) * std::sin(2. * x * y);
+
+  // Set this to as high as you want for greater accuracy
+  unsigned n = 3;
```

Comment:
According to the website you linked, this part of the formula should actually be an infinite series! 

What are the ramifications of truncating the series after an arbitrary n=3 terms, i.e. how quickly does the series converge and what is the error as a function of `n`? The algorithm in the linked question states that the number of terms needed to reach a desired tolerance is "fairly sensitive to" the argument, z.

I think at the very least there should be a tolerance (either hard-coded into the function or something specified by the user) that controls the number of iterations, possibly with a `max_its` to avoid infinite looping... there is even an algorithm in the code you linked.